tests/lifecycle/test_cache_adapter.py

```diff
@@ -138,3 +165,49 @@ def test_commit_nodes_history(hook: CacheAdapter):
     # need to reopen the hook cache
     with shelve.open(hook.cache_path) as cache:
         assert cache.get(CacheAdapter.nodes_history_key) == hook.nodes_history
+
+
+def test_partial_handling(hook: CacheAdapter, node_a_partial: node.Node):
```

Comment:
Yeah if the func-derivation is a function then this one is less necessary (although still nice to have), and we can test that/get more confidence.